controllers/pod.go

```diff
@@ -185,155 +160,227 @@ func (r *SingleClusterReconciler) quickRestart(
-		)
-
-		updatedPod := &corev1.Pod{}
-		err := r.Client.Get(
-			context.TODO(),
-			types.NamespacedName{Name: pod.Name, Namespace: pod.Namespace},
-			updatedPod,
-		)
+func (r *SingleClusterReconciler) restartPods(rackState RackState, podsToRestart []*corev1.Pod) reconcileResult {
+	var restartedPods []*corev1.Pod
+	for i := range podsToRestart {
+		pod := podsToRestart[i]
+		// Check if this pod needs restart
+		restartType, err := r.getRollingRestartTypePod(rackState, *pod)
 		if err != nil {
-			r.Log.Error(err, "Failed to get pod")
-			time.Sleep(time.Second * 5)
-			continue
+			return reconcileError(err)
 		}
 
-		if err := utils.CheckPodFailed(updatedPod); err != nil {
+		if restartType == QuickRestart {
+			if err := r.restartASDInPod(rackState, pod); err == nil {
+				continue
+			}
+			// If ASD restart fails then go ahead and restart the pod
+		}
+
+		if err := r.Client.Delete(context.TODO(), pod); err != nil {
+			r.Log.Error(err, "Failed to delete pod")
 			return reconcileError(err)
 		}
+		restartedPods = append(restartedPods, pod)
+
+		r.Log.V(1).Info("Pod deleted", "podName", pod.Name)
+	}
+	if len(restartedPods) > 0 {
+		return r.ensurePodsRunningAndReady(restartedPods)
+	}
+	return reconcileSuccess()
+}
+
+func (r *SingleClusterReconciler) ensurePodsRunningAndReady(podsToCheck []*corev1.Pod) reconcileResult {
+	podNames := getPodNames(podsToCheck)
+	readyPods := map[string]bool{}
+
+	const maxRetries = 6
+	const retryInterval = time.Second * 10
+
+	for i := 0; i < maxRetries; i++ {
+		r.Log.V(1).Info("Waiting for pods to be ready after delete", "pods", podNames)
+
+		for _, pod := range podsToCheck {
+			if readyPods[pod.Name] {
+				continue
+			}
 
-		if !utils.IsPodRunningAndReady(updatedPod) {
 			r.Log.V(1).Info(
-				"Waiting for pod to be ready", "podName", updatedPod.Name,
-				"status", updatedPod.Status.Phase, "DeletionTimestamp",
-				updatedPod.DeletionTimestamp,
+				"Waiting for pod to be ready", "podName", pod.Name,
+				"status", pod.Status.Phase, "DeletionTimestamp",
+				pod.DeletionTimestamp,
+			)
+
+			updatedPod := &corev1.Pod{}
+			podName := types.NamespacedName{Name: pod.Name, Namespace: pod.Namespace}
+			if err := r.Client.Get(context.TODO(), podName, updatedPod); err != nil {
+				return reconcileError(err)
+			}
+
+			if err := utils.CheckPodFailed(updatedPod); err != nil {
+				return reconcileError(err)
+			}
+
+			if !utils.IsPodRunningAndReady(updatedPod) {
+				break
+			}
+
+			readyPods[pod.Name] = true
+
+			r.Log.Info("Pod is restarted", "podName", updatedPod.Name)
+			r.Recorder.Eventf(r.aeroCluster, corev1.EventTypeNormal, "PodRestarted",
+				"[rack-%s] Restarted Pod %s", pod.Labels[asdbv1beta1.AerospikeRackIdLabel], pod.Name)
+		}
+
+		if len(readyPods) == len(podsToCheck) {
+			r.Log.Info(
+				"Pods are running and ready", "pods",
+				podNames,
 			)
-			time.Sleep(time.Second * 5)
+			return reconcileSuccess()
+		}
+		time.Sleep(retryInterval)
+	}
+
+	r.Log.Info(
```

Comment:
print what all pods were unable to come online and timeout happened.

Reply:
We do not check for all pods. Whenever we get an unready pod, we go out. To print all unready pods, we would need to check all pods.